### PR TITLE
Autofixer for ISC001

### LINKED ISSUE #4829: Autofix ISC001 (single-line-implicit-string-concatenation)
ISC001 could have an autofixer. It can be difficult to combine strings of different kinds, but it shouldn't be difficult to combine strings of the same kind, so, IMHO, it would be ok to fix only such cases.

### PR DESCRIPTION
**Pull request by tkukushkin**
## Summary

This PR adds autofixer for rule ISC001 in cases where both string literals are of the same kind and with same quotes (double / single).

Fixes #4829

## Test Plan

I added testcases with different combinations of string literals.

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/flake8_implicit_str_concat/ISC.py [hunk 1]
```diff
@@ -34,3 +34,15 @@
   b"abc"
   b"def"
 )
+
+_ = """a""" """b"""
```
Comment (MichaReiser) [inline crates/ruff/resources/test/fixtures/flake8_implicit_str_concat/ISC.py:38]: Can you add a test for joining multiline strings?

File: crates/ruff/src/rules/flake8_implicit_str_concat/rules/implicit.rs [hunk 2]
```diff
@@ -106,12 +112,95 @@ pub(crate) fn implicit(
...
+    Single,
+}
+
+fn get_quotes_style(text: &str) -> QuotesStyle {
+    if text.ends_with('\"') {
+        QuotesStyle::Double
+    } else {
+        QuotesStyle::Single
+    }
+}
```
Comment (MichaReiser) [inline crates/ruff/src/rules/flake8_implicit_str_concat/rules/implicit.rs:206]: The `str` module has some helpers to work with quotes. Can you check if you can re-use some of those?

https://github.com/charliermarsh/ruff/blob/466719247bc68c0b608bc85be341c5aa5e2a5ec2/crates/ruff_python_ast/src/str.rs#L22-L68

Comment (tkukushkin) [reply crates/ruff/src/rules/flake8_implicit_str_concat/rules/implicit.rs:206]: I don't know why, but these helpers don't work with f-strings.

File: crates/ruff/resources/test/fixtures/flake8_implicit_str_concat/ISC.py (hunk 1 above)
Comment (tkukushkin) [reply crates/ruff/resources/test/fixtures/flake8_implicit_str_concat/ISC.py:38]: Yes, good point, done!

**Pull request merged**